File: projects/Amazone_Get_Title_and_Price.ts
import { test } from '@playwright/test';

test('Amazon Laptop Prices', async ({ page }) => {

  // Open Amazon
  await page.goto('https://www.amazon.in/');

  // Search Laptop
  await page.fill('#twotabsearchtextbox', 'laptop');
  await page.keyboard.press('Enter');

  // Wait until products appear
  await page.waitForSelector('div[data-component-type="s-search-result"]');

  // Get all titles
  const titles = await page
    .locator('div[data-component-type="s-search-result"] h2')
    .allTextContents();

  // Get all prices
  const prices = await page
    .locator('.a-price-whole')
    .allTextContents();

  console.log("===== Laptop Names and Prices =====\n");

  // Print title and price
  for (let i = 0; i < Math.min(titles.length, prices.length); i++) {
    console.log(`${i + 1}. ${titles[i]}`);
    console.log(`Price: ₹${prices[i]}`);
    console.log("---------------------------");
  }

});

File: Playwright with Python/Amazone_Get_Title_and_Price.py
import pytest
import allure
from playwright.sync_api import sync_playwright


@allure.title("Playwright Mini Project - Amazon")
@allure.description("Print laptop items and prices from Amazon India")
@pytest.mark.itemsandprice
def test_print_items_price_amazon():

    with sync_playwright() as p:
        browser = p.chromium.launch(
            channel="msedge",
            headless=False
        )

        context = browser.new_context()
        page = context.new_page()

        page.goto("https://www.amazon.in/")
        page.wait_for_timeout(3000)

        # Search for laptop
        page.locator("#twotabsearchtextbox").fill("laptop")
        page.locator("#twotabsearchtextbox").press("Enter")

        page.wait_for_load_state("networkidle")

        # Get all product titles
        titles = page.locator(
            "//div[@data-component-type='s-search-result']//h2"
        ).all_text_contents()

        # Get all product prices
        prices = page.locator(
            "//div[@data-component-type='s-search-result']//span[@class='a-price-whole']"
        ).all_text_contents()

        print("\n===== Your Items and Prices on Amazon India =====\n")

        # Safely pair titles and prices
        for title, price in zip(titles, prices):
            print(f"{title} - ₹{price}")
            print()

        page.wait_for_timeout(5000)

        browser.close()
